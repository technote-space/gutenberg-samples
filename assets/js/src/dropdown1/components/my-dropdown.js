/* istanbul ignore next */
const { BlockFormatControls } = wp.blockEditor && wp.blockEditor.BlockEdit ? wp.blockEditor : wp.editor;
const { Toolbar, DropdownMenu } = wp.components;
import MyDropdownControls from './my-dropdown-controls';

const MyDropdown = () => <BlockFormatControls>
	<div className="editor-format-toolbar block-editor-format-toolbar">
		<Toolbar>
			<MyDropdownControls.Slot>
				{ fills => <DropdownMenu
					className='components-dropdown-button'
					icon='admin-customizer'
					popoverProps={ ( { position: 'bottom left' } ) }
					label='dropdown'
					controls={ fills.map( ( [ { props } ] ) => props ) }
				/> }
			</MyDropdownControls.Slot>
		</Toolbar>
	</div>
</BlockFormatControls>;

export default MyDropdown;
