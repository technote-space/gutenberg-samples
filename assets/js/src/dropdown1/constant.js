export const PLUGIN_NAME = 'dropdown1';
