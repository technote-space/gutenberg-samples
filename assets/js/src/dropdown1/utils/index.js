export { getToolbarButtonProps, createToolbarButton } from './toolbar-button';
export { getRichTextSetting } from './rich-text';