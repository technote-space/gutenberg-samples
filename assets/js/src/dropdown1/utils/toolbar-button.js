const { toggleFormat } = wp.richText;
const { ToolbarButton } = wp.components;

/**
 * @param {object} args args
 * @param {string} name name
 * @param {string} formatName format name
 * @returns {{onClick: onClick, icon: *, title: *, isActive: boolean}} toolbar button properties
 */
export const getToolbarButtonProps = ( { args, name, formatName } ) => ( {
	icon: 'admin-customizer',
	title: <div className={ name }>
		{ name }
	</div>,
	onClick: () => {
		args.onChange( toggleFormat( args.value, {
			type: formatName,
		} ) );
	},
	isActive: args.isActive,
} );

/**
 * @param {object} args args
 * @param {string} name name
 * @param {string} formatName format name
 * @returns {*} toolbar button
 */
export const createToolbarButton = ( { args, name, formatName } ) => <ToolbarButton { ...getToolbarButtonProps( { args, name, formatName } ) } />;