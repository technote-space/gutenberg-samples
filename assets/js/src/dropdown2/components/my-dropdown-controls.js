const { createSlotFill } = wp.components;

/**
 * @param {string} group group
 * @returns {*} slot fill
 * @constructor
 */
const MyDropdownControls = group => {
	const { Fill, Slot } = createSlotFill( 'MyDropdownControls.' + group );
	Fill.Slot = Slot;
	return Fill;
};

export default MyDropdownControls;
