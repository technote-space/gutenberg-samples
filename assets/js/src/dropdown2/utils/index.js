export { getToolbarButtonProps, registerGroupedFormatType } from './toolbar-button';
export { getRichTextSetting } from './rich-text';