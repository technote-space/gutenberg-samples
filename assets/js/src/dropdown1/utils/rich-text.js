const { Fragment } = wp.element;

import { PLUGIN_NAME } from '../../common/constant';
import MyDropdown from '../components/my-dropdown';
import MyDropdownControls from '../components/my-dropdown-controls';

/**
 * @param {string} name name
 * @param {number} index index
 * @param {function} create create component function
 * @param {object} setting setting
 * @returns {array} setting
 */
export const getRichTextSetting = ( { name, create, setting = {} }, index ) => {
	const formatName = PLUGIN_NAME + '/' + name;
	const component = args => <MyDropdownControls>
		{ create( { args, name, formatName } ) }
	</MyDropdownControls>;

	setting.title = setting.title || name;
	setting.tagName = setting.tagName || 'span';
	setting.className = setting.className || name;
	setting.edit = args => {
		if ( ! index ) {
			return <Fragment>
				{ component( args ) }
				<MyDropdown/>
			</Fragment>;
		}
		return component( args );
	};
	return [ formatName, setting ];
};
