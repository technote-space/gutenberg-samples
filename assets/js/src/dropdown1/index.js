const { registerFormatType } = wp.richText;

import { createToolbarButton, getRichTextSetting } from './utils';

[
	{
		name: 'test1',
		create: createToolbarButton,
	},
	{
		name: 'test2',
		create: createToolbarButton,
	},
	{
		name: 'test3',
		create: createToolbarButton,
	},
].forEach( ( { name, create, setting = {} }, index ) => registerFormatType( ...getRichTextSetting( { name, create, setting }, index ) ) );
