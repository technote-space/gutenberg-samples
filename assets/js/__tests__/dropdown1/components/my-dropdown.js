/* eslint-disable no-magic-numbers */
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MyDropdown, MyDropdownControls } from '../../../src/dropdown1/components';
import { createToolbarButton } from '../../../src/dropdown1/utils';

const { BlockEdit, BlockFormatControls } = wp.blockEditor;
const { SlotFillProvider }               = wp.components;
const { createHigherOrderComponent }     = wp.compose;
const { Fragment }                       = wp.element;
const { addFilter, removeFilter }        = wp.hooks;
const { create }                         = wp.richText;

beforeAll(() => {
  addFilter('editor.BlockEdit', 'components-test/dropdown1-test', createHigherOrderComponent(BlockEdit => props =>
    <Fragment>
      <BlockEdit {...props}/>
      <MyDropdown/>
      <MyDropdownControls>
        {createToolbarButton({ args: { isActive: true }, name: 'test1', formatName: 'test1f' })}
      </MyDropdownControls>
      <MyDropdownControls>
        {createToolbarButton({ args: { isActive: false }, name: 'test2', formatName: 'test2f' })}
      </MyDropdownControls>
    </Fragment>,
  ));
});

afterAll(() => {
  removeFilter('editor.BlockEdit', 'components-test/dropdown1-test');
});

describe('MyDropdown', () => {
  it('should render MyDropdown', () => {
    const wrapper = mount(
      <SlotFillProvider>
        <BlockFormatControls.Slot/>
        <BlockEdit
          name="core/quote"
          isSelected={true}
          attributes={({
            className: 'test-block-edit',
            content: create({
              text: 'test',
              start: 0,
              end: 1,
              formats: [[], [], [], []],
            }),
          })}
        />
      </SlotFillProvider>,
    );

    expect(toJson(wrapper, { mode: 'deep' })).toMatchSnapshot('render');

    expect(wrapper.find('.components-button.components-dropdown-menu__toggle').hostNodes()).toHaveLength(1);
    expect(wrapper.find('.components-button.components-dropdown-menu__toggle.is-opened').hostNodes()).toHaveLength(0);
    wrapper.find('.components-button.components-dropdown-menu__toggle').hostNodes().simulate('click');

    expect(toJson(wrapper, { mode: 'deep' })).toMatchSnapshot('click');
    expect(wrapper.find('.components-button.components-dropdown-menu__toggle.is-opened').hostNodes()).toHaveLength(1);
    expect(wrapper.find('.components-button.components-dropdown-menu__menu-item .test1').hostNodes()).toHaveLength(1);
    expect(wrapper.find('.components-button.components-dropdown-menu__menu-item .test2').hostNodes()).toHaveLength(1);
  });
});
