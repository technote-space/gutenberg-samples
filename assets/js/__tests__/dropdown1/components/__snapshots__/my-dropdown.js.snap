// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyDropdown should render MyDropdown: click 1`] = `
Array [
  <div
    className="editor-format-toolbar block-editor-format-toolbar"
    key="2---0/.0"
  >
    <div
      className="components-toolbar"
    >
      <div
        className="components-dropdown components-dropdown-menu components-dropdown-button"
      >
        <ForwardRef(Button)
          aria-expanded={true}
          aria-haspopup="true"
          className="components-dropdown-menu__toggle is-opened"
          icon="admin-customizer"
          label="dropdown"
          onClick={[Function]}
          onKeyDown={[Function]}
          showTooltip={true}
        >
          <button
            aria-expanded={true}
            aria-haspopup="true"
            aria-label="dropdown"
            className="components-button components-dropdown-menu__toggle is-opened has-icon"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            type="button"
          >
            <svg
              aria-hidden="true"
              className="dashicon dashicons-admin-customizer"
              focusable="false"
              height={20}
              role="img"
              viewBox="0 0 20 20"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18.33 3.57s.27-.8-.31-1.36c-.53-.52-1.22-.24-1.22-.24-.61.3-5.76 3.47-7.67 5.57-.86.96-2.06 3.79-1.09 4.82.92.98 3.96-.17 4.79-1 2.06-2.06 5.21-7.17 5.5-7.79zM1.4 17.65c2.37-1.56 1.46-3.41 3.23-4.64.93-.65 2.22-.62 3.08.29.63.67.8 2.57-.16 3.46-1.57 1.45-4 1.55-6.15.89z"
              />
            </svg>
          </button>
        </ForwardRef(Button)>
        <span>
          <div
            onKeyDown={[Function]}
            tabIndex="-1"
          >
            <div
              onBlur={[Function]}
              onFocus={[Function]}
            >
              <div
                onBlur={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
              >
                <ForwardRef
                  className="components-popover components-dropdown-menu__popover"
                  onKeyDown={[Function]}
                >
                  <div
                    className="components-popover components-dropdown-menu__popover"
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                  >
                    <div
                      className="components-popover__content"
                      tabIndex="-1"
                    >
                      <ForwardRef(NavigableMenu)
                        aria-label="dropdown"
                        className="components-dropdown-menu__menu"
                        role="menu"
                      >
                        <ForwardRef(NavigableContainer)
                          aria-label="dropdown"
                          aria-orientation="vertical"
                          className="components-dropdown-menu__menu"
                          eventToOffset={[Function]}
                          onlyBrowserTabstops={false}
                          role="menu"
                          stopNavigationEvents={true}
                        >
                          <div
                            aria-label="dropdown"
                            aria-orientation="vertical"
                            className="components-dropdown-menu__menu"
                            role="menu"
                          >
                            <ForwardRef(Button)
                              className="components-dropdown-menu__menu-item is-active"
                              icon="admin-customizer"
                              key="0,0"
                              onClick={[Function]}
                              role="menuitem"
                            >
                              <button
                                className="components-button components-dropdown-menu__menu-item is-active has-text has-icon"
                                onClick={[Function]}
                                role="menuitem"
                                type="button"
                              >
                                <svg
                                  aria-hidden="true"
                                  className="dashicon dashicons-admin-customizer"
                                  focusable="false"
                                  height={20}
                                  role="img"
                                  viewBox="0 0 20 20"
                                  width={20}
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M18.33 3.57s.27-.8-.31-1.36c-.53-.52-1.22-.24-1.22-.24-.61.3-5.76 3.47-7.67 5.57-.86.96-2.06 3.79-1.09 4.82.92.98 3.96-.17 4.79-1 2.06-2.06 5.21-7.17 5.5-7.79zM1.4 17.65c2.37-1.56 1.46-3.41 3.23-4.64.93-.65 2.22-.62 3.08.29.63.67.8 2.57-.16 3.46-1.57 1.45-4 1.55-6.15.89z"
                                  />
                                </svg>
                                <div
                                  className="test1"
                                >
                                  test1
                                </div>
                              </button>
                            </ForwardRef(Button)>
                            <ForwardRef(Button)
                              className="components-dropdown-menu__menu-item"
                              icon="admin-customizer"
                              key="0,1"
                              onClick={[Function]}
                              role="menuitem"
                            >
                              <button
                                className="components-button components-dropdown-menu__menu-item has-text has-icon"
                                onClick={[Function]}
                                role="menuitem"
                                type="button"
                              >
                                <svg
                                  aria-hidden="true"
                                  className="dashicon dashicons-admin-customizer"
                                  focusable="false"
                                  height={20}
                                  role="img"
                                  viewBox="0 0 20 20"
                                  width={20}
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M18.33 3.57s.27-.8-.31-1.36c-.53-.52-1.22-.24-1.22-.24-.61.3-5.76 3.47-7.67 5.57-.86.96-2.06 3.79-1.09 4.82.92.98 3.96-.17 4.79-1 2.06-2.06 5.21-7.17 5.5-7.79zM1.4 17.65c2.37-1.56 1.46-3.41 3.23-4.64.93-.65 2.22-.62 3.08.29.63.67.8 2.57-.16 3.46-1.57 1.45-4 1.55-6.15.89z"
                                  />
                                </svg>
                                <div
                                  className="test2"
                                >
                                  test2
                                </div>
                              </button>
                            </ForwardRef(Button)>
                          </div>
                        </ForwardRef(NavigableContainer)>
                      </ForwardRef(NavigableMenu)>
                    </div>
                  </div>
                </ForwardRef>
              </div>
            </div>
          </div>
        </span>
      </div>
    </div>
  </div>,
  Array [
    Array [
      "",
      <blockquote
        className="wp-block-quote test-block-edit"
      >
        <ForwardRef(RichTextWrapper)
          __unstableOnSplitMiddle={[Function]}
          identifier="value"
          multiline={true}
          onChange={[Function]}
          onRemove={[Function]}
          onSplit={[Function]}
          placeholder="Write quote…"
        >
          <ForwardRef
            __unstableDidAutomaticChange={false}
            __unstableInputRule={[Function]}
            __unstableIsCaretWithinFormattedText={false}
            __unstableIsSelected={false}
            __unstableMarkAutomaticChange={[Function]}
            __unstableMultilineTag="p"
            __unstableOnCreateUndoLevel={[Function]}
            __unstableOnEnterFormattedText={[Function]}
            __unstableOnExitFormattedText={[Function]}
            blockIsSelected={true}
            className="block-editor-rich-text__editable"
            disabled={false}
            identifier="value"
            onChange={[Function]}
            onDelete={[Function]}
            onEnter={[Function]}
            onPaste={[Function]}
            onSelectionChange={[Function]}
            placeholder="Write quote…"
          >
            Array [
              "",
              "",
              <div
                aria-activedescendant={null}
                aria-label="Write quote…"
                aria-multiline={true}
                aria-owns={null}
                className="rich-text block-editor-rich-text__editable"
                contentEditable={true}
                onBlur={[Function]}
                onCompositionEnd={[Function]}
                onCompositionStart={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
                onPaste={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                role="textbox"
                style={
                  Object {
                    "whiteSpace": "pre-wrap",
                  }
                }
                suppressContentEditableWarning={true}
              />,
            ]
          </ForwardRef>
        </ForwardRef(RichTextWrapper)>
        <ForwardRef(RichTextWrapper)
          __unstableMobileNoFocusOnMount={true}
          className="wp-block-quote__citation"
          identifier="citation"
          onChange={[Function]}
          placeholder="Write citation…"
        >
          <ForwardRef
            __unstableDidAutomaticChange={false}
            __unstableInputRule={[Function]}
            __unstableIsCaretWithinFormattedText={false}
            __unstableIsSelected={false}
            __unstableMarkAutomaticChange={[Function]}
            __unstableMobileNoFocusOnMount={true}
            __unstableOnCreateUndoLevel={[Function]}
            __unstableOnEnterFormattedText={[Function]}
            __unstableOnExitFormattedText={[Function]}
            blockIsSelected={true}
            className="block-editor-rich-text__editable wp-block-quote__citation"
            disabled={false}
            identifier="citation"
            onChange={[Function]}
            onDelete={[Function]}
            onEnter={[Function]}
            onPaste={[Function]}
            onSelectionChange={[Function]}
            placeholder="Write citation…"
          >
            Array [
              "",
              "",
              <div
                aria-activedescendant={null}
                aria-label="Write citation…"
                aria-multiline={true}
                aria-owns={null}
                className="rich-text block-editor-rich-text__editable wp-block-quote__citation"
                contentEditable={true}
                onBlur={[Function]}
                onCompositionEnd={[Function]}
                onCompositionStart={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
                onPaste={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                role="textbox"
                style={
                  Object {
                    "whiteSpace": "pre-wrap",
                  }
                }
                suppressContentEditableWarning={true}
              />,
            ]
          </ForwardRef>
        </ForwardRef(RichTextWrapper)>
      </blockquote>,
    ],
    "",
    "",
    "",
  ],
]
`;

exports[`MyDropdown should render MyDropdown: render 1`] = `
Array [
  <div
    className="editor-format-toolbar block-editor-format-toolbar"
    key="2---0/.0"
  >
    <div
      className="components-toolbar"
    >
      <div
        className="components-dropdown components-dropdown-menu components-dropdown-button"
      >
        <ForwardRef(Button)
          aria-expanded={false}
          aria-haspopup="true"
          className="components-dropdown-menu__toggle"
          icon="admin-customizer"
          label="dropdown"
          onClick={[Function]}
          onKeyDown={[Function]}
          showTooltip={true}
        >
          <button
            aria-expanded={false}
            aria-haspopup="true"
            aria-label="dropdown"
            className="components-button components-dropdown-menu__toggle has-icon"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            type="button"
          >
            <svg
              aria-hidden="true"
              className="dashicon dashicons-admin-customizer"
              focusable="false"
              height={20}
              role="img"
              viewBox="0 0 20 20"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18.33 3.57s.27-.8-.31-1.36c-.53-.52-1.22-.24-1.22-.24-.61.3-5.76 3.47-7.67 5.57-.86.96-2.06 3.79-1.09 4.82.92.98 3.96-.17 4.79-1 2.06-2.06 5.21-7.17 5.5-7.79zM1.4 17.65c2.37-1.56 1.46-3.41 3.23-4.64.93-.65 2.22-.62 3.08.29.63.67.8 2.57-.16 3.46-1.57 1.45-4 1.55-6.15.89z"
              />
            </svg>
          </button>
        </ForwardRef(Button)>
      </div>
    </div>
  </div>,
  Array [
    Array [
      "",
      <blockquote
        className="wp-block-quote test-block-edit"
      >
        <ForwardRef(RichTextWrapper)
          __unstableOnSplitMiddle={[Function]}
          identifier="value"
          multiline={true}
          onChange={[Function]}
          onRemove={[Function]}
          onSplit={[Function]}
          placeholder="Write quote…"
        >
          <ForwardRef
            __unstableDidAutomaticChange={false}
            __unstableInputRule={[Function]}
            __unstableIsCaretWithinFormattedText={false}
            __unstableIsSelected={false}
            __unstableMarkAutomaticChange={[Function]}
            __unstableMultilineTag="p"
            __unstableOnCreateUndoLevel={[Function]}
            __unstableOnEnterFormattedText={[Function]}
            __unstableOnExitFormattedText={[Function]}
            blockIsSelected={true}
            className="block-editor-rich-text__editable"
            disabled={false}
            identifier="value"
            onChange={[Function]}
            onDelete={[Function]}
            onEnter={[Function]}
            onPaste={[Function]}
            onSelectionChange={[Function]}
            placeholder="Write quote…"
          >
            Array [
              "",
              "",
              <div
                aria-activedescendant={null}
                aria-label="Write quote…"
                aria-multiline={true}
                aria-owns={null}
                className="rich-text block-editor-rich-text__editable"
                contentEditable={true}
                onBlur={[Function]}
                onCompositionEnd={[Function]}
                onCompositionStart={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
                onPaste={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                role="textbox"
                style={
                  Object {
                    "whiteSpace": "pre-wrap",
                  }
                }
                suppressContentEditableWarning={true}
              />,
            ]
          </ForwardRef>
        </ForwardRef(RichTextWrapper)>
        <ForwardRef(RichTextWrapper)
          __unstableMobileNoFocusOnMount={true}
          className="wp-block-quote__citation"
          identifier="citation"
          onChange={[Function]}
          placeholder="Write citation…"
        >
          <ForwardRef
            __unstableDidAutomaticChange={false}
            __unstableInputRule={[Function]}
            __unstableIsCaretWithinFormattedText={false}
            __unstableIsSelected={false}
            __unstableMarkAutomaticChange={[Function]}
            __unstableMobileNoFocusOnMount={true}
            __unstableOnCreateUndoLevel={[Function]}
            __unstableOnEnterFormattedText={[Function]}
            __unstableOnExitFormattedText={[Function]}
            blockIsSelected={true}
            className="block-editor-rich-text__editable wp-block-quote__citation"
            disabled={false}
            identifier="citation"
            onChange={[Function]}
            onDelete={[Function]}
            onEnter={[Function]}
            onPaste={[Function]}
            onSelectionChange={[Function]}
            placeholder="Write citation…"
          >
            Array [
              "",
              "",
              <div
                aria-activedescendant={null}
                aria-label="Write citation…"
                aria-multiline={true}
                aria-owns={null}
                className="rich-text block-editor-rich-text__editable wp-block-quote__citation"
                contentEditable={true}
                onBlur={[Function]}
                onCompositionEnd={[Function]}
                onCompositionStart={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
                onPaste={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                role="textbox"
                style={
                  Object {
                    "whiteSpace": "pre-wrap",
                  }
                }
                suppressContentEditableWarning={true}
              />,
            ]
          </ForwardRef>
        </ForwardRef(RichTextWrapper)>
      </blockquote>,
    ],
    "",
    "",
    "",
  ],
]
`;
